target_chains/ethereum/contracts/contracts/executor/Executor.sol

```diff
@@ -0,0 +1,161 @@
+
+    // Magic is `PTGM` encoded as a 4 byte data: Pyth Governance Message
+    // TODO: it's annoying that we can't import this from PythGovernanceInstructions
+    uint32 constant MAGIC = 0x5054474d;
+
+    PythGovernanceInstructions.GovernanceModule constant MODULE =
+        PythGovernanceInstructions.GovernanceModule.EvmExecutor;
+
+    // Instruction indicating that the executor contract on
+    // targetChainId at executorAddress should call the contract at callAddress
+    // with the provided callData
+    struct GovernanceInstruction {
+        PythGovernanceInstructions.GovernanceModule module;
+        ExecutorAction action;
+        uint16 targetChainId;
+        // The address of the specific executor that should perform the call.
+        // This argument is included to support multiple Executors on the same blockchain.
+        address executorAddress;
+        address callAddress;
+        bytes callData;
+    }
+
+    // We have different actions here for potential future extensibility
+    enum ExecutorAction {
+        Execute // 0
+    }
+
+    IWormhole private wormhole;
+    uint64 private lastExecutedSequence;
+    uint16 private chainId;
+
+    uint16 private ownerEmitterChainId;
+    bytes32 private ownerEmitterAddress;
+
+    constructor(
+        address _wormhole,
+        uint64 _lastExecutedSequence,
+        uint16 _chainId,
+        uint16 _ownerEmitterChainId,
+        bytes32 _ownerEmitterAddress
+    ) {
+        wormhole = IWormhole(_wormhole);
+        lastExecutedSequence = _lastExecutedSequence;
+        chainId = _chainId;
+        ownerEmitterChainId = _ownerEmitterChainId;
+        ownerEmitterAddress = _ownerEmitterAddress;
+    }
+
+    // Execute the contract call in the provided wormhole message.
+    // The argument should be the bytes of a valid wormhole message
+    // whose payload is a serialized GovernanceInstruction.
+    function execute(
+        bytes memory encodedVm
+    ) public returns (bytes memory response) {
+        IWormhole.VM memory vm = verifyGovernanceVM(encodedVm);
+
+        GovernanceInstruction memory gi = parseGovernanceInstruction(
+            vm.payload
+        );
+
+        if (gi.targetChainId != chainId && gi.targetChainId != 0)
+            revert ExecutorErrors.InvalidGovernanceTarget();
+
+        if (
+            gi.action != ExecutorAction.Execute ||
+            gi.executorAddress != address(this)
+        ) revert ExecutorErrors.DeserializationError();
+
+        bool success;
+        (success, response) = address(gi.callAddress).call(gi.callData);
+
+        // Check if the call was successful or not.
+        if (!success) {
+            // If there is return data, the delegate call reverted with a reason or a custom error, which we bubble up.
+            if (response.length > 0) {
+                assembly {
+                    let returndata_size := mload(response)
+                    revert(add(32, response), returndata_size)
```

Comment:
I copied this logic from another proxy contract https://github.com/PaulRBerg/prb-proxy

Reply:
Hmm in my mind it should be add(20, response). Let me see the other one.

Reply:
Oh no my bad, 32 is a word size.

Reply:
Maybe add a documentation to to explain that the first word of response is the length, so when we call revert we add response pointer by a word (32 bytes) to give the pointer to the beginning of the revert data and pass the size as the second argument.